internal/cli/cmd/playground/base.go

```diff
@@ -60,8 +61,17 @@ func (o *baseOptions) validate() error {
 	}
 
 	// check the kbcli version
-	if o.prevCluster != nil && o.prevCluster.KbcliVersion != "" && o.prevCluster.KbcliVersion != version.GetVersion() {
-		return fmt.Errorf("the playground cluster is created by kbcli %s, but the current kbcli version is %s, please use the same version of kbcli to manage the playground cluster", o.prevCluster.KbcliVersion, version.GetVersion())
+	if o.prevCluster != nil && o.prevCluster.KbcliVersion != "" {
+		prev, err := gv.NewVersion(o.prevCluster.KbcliVersion)
+		fmt.Printf("failed to parse previous kbcli version %s, %v\n", o.prevCluster.KbcliVersion, err)
+
+		cur, err := gv.NewVersion(version.GetVersion())
+		fmt.Printf("failed to parse current kbcli version %s, %v\n", version.GitVersion, err)
+
+		// compare core version
+		if prev != nil && cur != nil && !prev.Core().Equal(cur.Core()) {
+			fmt.Printf("the playground cluster is created by kbcli %s, but the current kbcli version is %s, please use the same version of kbcli to manage the playground cluster", o.prevCluster.KbcliVersion, version.GetVersion())
```

Comment:
The playground use the cloud-provider directory that is based on the kbcli version, such as cloud-provider-0.5.0. If kbcli version is different, it can't find the correct directory, so playground better to record the cloud-provider path and use it when destroy.